x-pack/plugins/integration_assistant/server/graphs/unstructured/validate.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { UnstructuredLogState } from '../../types';
+import type { GrokResult, HandleUnstructuredNodeParams } from './types';
+import { testPipeline } from '../../util';
+import { onFailure } from './constants';
+import { createGrokProcessor } from '../../util/processors';
+
+export async function handleUnstructuredValidate({
+  state,
+  client,
+}: HandleUnstructuredNodeParams): Promise<Partial<UnstructuredLogState>> {
+  const grokPatterns = state.grokPatterns;
+  const grokProcessor = createGrokProcessor(grokPatterns);
+  const pipeline = { processors: grokProcessor, on_failure: [onFailure] };
+
+  const { pipelineResults, errors } = (await testPipeline(state.logSamples, pipeline, client)) as {
+    pipelineResults: GrokResult[];
+    errors: object[];
+  };
+
+  if (errors.length > 0) {
+    return { errors, lastExecutedChain: 'unstructured_validate' };
+  }
+
+  const jsonSamples: string[] = pipelineResults.map((entry) => JSON.stringify(entry));
+  const additionalProcessors = state.additionalProcessors;
+  additionalProcessors.push(grokProcessor[0]);
```

Comment:
why do we need `additionalProcessors.push(grokProcessor[0]);`?

Reply:
Since , we are generating `grok` processor here and then we proceed to ECS Mapping , we need this processor as the first processor that handles the logs to convert them to json before being handled by other processors. Hence we pass this around.